etcdserver/api/v3rpc/auth.go

```diff
@@ -28,8 +31,21 @@ func NewAuthServer(s *etcdserver.EtcdServer) *AuthServer {
 	return &AuthServer{authenticator: s}
 }
 
+func knownAuthError(err error) bool {
```

Comment:
use to grpc error in key.go (probably copy that function into a separate file since it now is used by mutiple files)? we need to modify it by adding the new error defined in error.go

```
if err == auth.UserExist
    return ErrUserAlreadyExist
```